### ISSUE TITLE
Prepare pull requests to get my code into Strudel

### ISSUE THREAD

**Issue by gogins** [gogins/cloud-5#33]
- [x] Reduce patches of Strudel code to the absolute minimum. _That already seems to be the case._
- [x] Determine if any of my patches are no longer necessary (run the Logistic example and _Cloud Music No. 9_). _That does not seem to be the case._
- [ ] PR: API for embedded Strudel, as a separate module.
- [ ] PR: FIx multiple file separators.
- [x] PR? StatefulPatterns. _Not doing._

**Comment by gogins** [comment]
Summary for FR:

- The project I'm working on this one, cloud-5, which is the library and toolkit I use to make my Web site of online, always-on music, cloud-music.
- cloud-5 is an all-HTML5 computer music that runs only in standard Web browsers, either locally hosted or on the WWW.
- In this system, a musical composition is a single HTML file that can run Csound, Strudel, and my C++/WebAssembly library for algorithmic composition, CsoundAC.
- We have already succeeded in getting Csound to play notes from Strudel.
- I have succeeded in writing a JavaScript module, statefulpatterns.mjs, that can carry objects with state from CsoundAC, or any other object with state, into Strudel Patterns. This is done by subclassing the StatefulPatterns class, which automatically registers all class methods as Strudel Patterns; each method acts both as a trigger and as a plain value.
- I have had no luck synchronizing this state with Strudel haps _except_ by using a silent WebAudio node. That is what `csoundn` in `csoundac.mjs` does. I.e., `csoundn` is scheduled at exactly the same time as the silent WebAudio output. It does not use a trigger as `csound` does. I surmise that only at the point of real-time audio output can the queries of Strudel actually be matched with stateful values. Also, the scheduling only works if `csoundn` has access to timing information from Cyclist, so I exposed Cyclist as a singleton in `globalThis.__cyclist__.`
- The `pianoroll` also doesn't work out of the box. Either it doesn't show up, or it shows a flat line of notes that jumps at every cycle boundary. I added also a global Hap buffer in `globalThis.__haps_from_outputs__`; `csoundn` pushes all its haps onto this buffer.
- Then `pianoroll` pushes the Haps from `csoundn` onto this global buffer, which is full of extraneous Haps from queries. The buffer is filtered and assigned to the regular `haps` object in `pianoroll.` Which then displays properly. If I knew more about the innards of Strudel perhaps I could get rid of these extraneous Haps. However, they don't seem to cause a big performance hit.
